python2.7 pytest-sugar fails to install due to test failures

## Issue description

`nixpkgs.python27Packages.pytest-sugar` has a test failure and won't install.

### Steps to reproduce

```
adam: [~] $ nix-env -iA nixpkgs.python27Packages.pytest-sugar
installing ‘python2.7-pytest-sugar-0.9.1’
these derivations will be built:
  /nix/store/xjc1ywcjw2ndy6m2lspynq4hjldwsajy-python2.7-pytest-sugar-0.9.1.drv
building path(s) ‘/nix/store/xk8amwv8sq9arvi301341m6vmb889h5x-python2.7-pytest-sugar-0.9.1’
unpacking sources
unpacking source archive /nix/store/zhy7vvln4hg9mpjwnx8x5h570r9x4m8d-pytest-sugar-0.9.1.tar.gz
source root is pytest-sugar-0.9.1
setting SOURCE_DATE_EPOCH to timestamp 1517923935 of file pytest-sugar-0.9.1/setup.cfg
patching sources
configuring
building
running bdist_wheel
running build
running build_py
creating build
creating build/lib
copying pytest_sugar.py -> build/lib
running egg_info
writing requirements to pytest_sugar.egg-info/requires.txt
writing pytest_sugar.egg-info/PKG-INFO
writing top-level names to pytest_sugar.egg-info/top_level.txt
writing dependency_links to pytest_sugar.egg-info/dependency_links.txt
writing entry points to pytest_sugar.egg-info/entry_points.txt
reading manifest file 'pytest_sugar.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'pytest_sugar.egg-info/SOURCES.txt'
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
copying build/lib/pytest_sugar.py -> build/bdist.linux-x86_64/wheel
running install_egg_info
Copying pytest_sugar.egg-info to build/bdist.linux-x86_64/wheel/pytest_sugar-0.9.1-py2.7.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/pytest_sugar-0.9.1.dist-info/WHEEL
creating '/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-sugar-0.9.1/dist/pytest_sugar-0.9.1-py2.py3-none-any.whl' and adding '.' to it
adding 'pytest_sugar.py'
adding 'pytest_sugar-0.9.1.dist-info/DESCRIPTION.rst'
adding 'pytest_sugar-0.9.1.dist-info/entry_points.txt'
adding 'pytest_sugar-0.9.1.dist-info/metadata.json'
adding 'pytest_sugar-0.9.1.dist-info/top_level.txt'
adding 'pytest_sugar-0.9.1.dist-info/WHEEL'
adding 'pytest_sugar-0.9.1.dist-info/METADATA'
adding 'pytest_sugar-0.9.1.dist-info/RECORD'
installing
/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-sugar-0.9.1/dist /tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-sugar-0.9.1
Processing ./pytest_sugar-0.9.1-py2.py3-none-any.whl
Requirement already satisfied: termcolor>=1.1.0 in /nix/store/2nm5g4vp25g1493g140bkkrj5nyjfqhb-python2.7-termcolor-1.1.0/lib/python2.7/site-packages (from pytest-sugar==0.9.1)
Requirement already satisfied: pytest>=2.9 in /nix/store/5dzmpz9wapsvbjy4pcslbg32dddgsxxq-python2.7-pytest-3.5.0/lib/python2.7/site-packages (from pytest-sugar==0.9.1)
Requirement already satisfied: more-itertools>=4.0.0 in /nix/store/4md366zp5pbaiv1im47h13wixlrvd8q8-python2.7-more-itertools-4.0.1/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Requirement already satisfied: setuptools in /nix/store/hf7ls1fdzs5qpr1kl083agyhhv15wfa7-python2.7-bootstrapped-pip-9.0.3/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Requirement already satisfied: attrs>=17.4.0 in /nix/store/yv2ygk0aiwlvf2jd6igp09w2ln83jkcy-python2.7-attrs-17.4.0/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Requirement already satisfied: py>=1.5.0 in /nix/store/ylqnk4kwayw46hmxabhfsxgp3ifz85lh-python2.7-py-1.5.3/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Requirement already satisfied: six>=1.10.0 in /nix/store/d2dgyy3fwrxip2l63lhjiwx707gm1v6w-python2.7-six-1.11.0/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Requirement already satisfied: funcsigs; python_version < "3.0" in /nix/store/vyxph66an49vpv9q9np1hccp7y3kywxz-python2.7-funcsigs-1.0.2/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Requirement already satisfied: pluggy<0.7,>=0.5 in /nix/store/qxrw0hrcjmw3vd9sh654q2jscqjbxi7k-python2.7-pluggy-0.6.0/lib/python2.7/site-packages (from pytest>=2.9->pytest-sugar==0.9.1)
Installing collected packages: pytest-sugar
Successfully installed pytest-sugar-0.9.1
/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-sugar-0.9.1
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/xk8amwv8sq9arvi301341m6vmb889h5x-python2.7-pytest-sugar-0.9.1
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/xk8amwv8sq9arvi301341m6vmb889h5x-python2.7-pytest-sugar-0.9.1/lib 
patching script interpreter paths in /nix/store/xk8amwv8sq9arvi301341m6vmb889h5x-python2.7-pytest-sugar-0.9.1
checking for references to /tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0 in /nix/store/xk8amwv8sq9arvi301341m6vmb889h5x-python2.7-pytest-sugar-0.9.1...
running install tests
============================= test session starts ==============================
platform linux2 -- Python 2.7.14, pytest-3.5.0, py-1.5.3, pluggy-0.6.0
rootdir: /tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-sugar-0.9.1, inifile:
plugins: sugar-0.9.1
collected 27 items

test_sugar.py \x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 11%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 22%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 33%]
\x1b[32m\u2713\x1b[0ms\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 44%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 55%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 66%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m [ 77%]
\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0ms\x1b[32m\u2713\x1b[0m [ 88%]
\x1b[32m\u2713\x1b[0ms\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[32m\u2713\x1b[0m\x1b[31m\u2a2f\x1b[0m\x1b[32m\u2713\x1b[0m [100%]

=================================== FAILURES ===================================
___________________ TestTerminalReporter.test_doctest_lineno ___________________

self = <test_sugar.TestTerminalReporter object at 0x7ffff3116490>
testdir = <Testdir local('/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0')>

    def test_doctest_lineno(self, testdir):
        """ Test location reported for doctest-modules """
    
        testdir.makepyfile(
            """
                def foobar():
                    '''
                    >>> foobar()
                    '''
                    raise NotImplementedError
                """
        )
        result = testdir.runpytest('--force-sugar', '--doctest-module')
    
        assert result.ret == 1, result.stderr.str()
        result.stdout.fnmatch_lines([
            'UNEXPECTED EXCEPTION: NotImplementedError()',
            '*test_doctest_lineno.py:3: UnexpectedException',
            'Results*:',
>           '*-*test_doctest_lineno.py*:3*',
        ])

../../../pytest-sugar-0.9.1/test_sugar.py:514: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_pytest.pytester.LineMatcher object at 0x7ffff310bc90>
lines2 = ['UNEXPECTED EXCEPTION: NotImplementedError()', '*test_doctest_lineno.py:3: UnexpectedException', 'Results*:', '*-*test_doctest_lineno.py*:3*']

    def fnmatch_lines(self, lines2):
        """Search captured text for matching lines using ``fnmatch.fnmatch``.
    
            The argument is a list of lines which have to match and can use glob
            wildcards.  If they do not match a pytest.fail() is called.  The
            matches and non-matches are also printed on stdout.
    
            """
>       self._match_lines(lines2, fnmatch, 'fnmatch')
E       Failed: nomatch: 'UNEXPECTED EXCEPTION: NotImplementedError()'
E           and: u'Test session starts (platform: linux2, Python 2.7.14, pytest 3.5.0, pytest-sugar 0.9.1)'
E           and: u'rootdir: /tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0, inifile:'
E           and: u'plugins: sugar-0.9.1'
E           and: u''
E           and: u''
E           and: u'\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015 [doctest] test_doctest_lineno.foobar \u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015'
E           and: u'002 '
E           and: u'003                 >>> foobar()'
E       exact match: 'UNEXPECTED EXCEPTION: NotImplementedError()'
E       nomatch: '*test_doctest_lineno.py:3: UnexpectedException'
E           and: u'Traceback (most recent call last):'
E           and: u''
E           and: u'  File "/nix/store/l9j2jsc9flrbmpf799nw9wdq59gpwms8-python-2.7.14/lib/python2.7/doctest.py", line 1315, in __run'
E           and: u'    compileflags, 1) in test.globs'
E           and: u''
E           and: u'  File "<doctest test_doctest_lineno.foobar[0]>", line 1, in <module>'
E           and: u''
E           and: u'  File "/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0/test_doctest_lineno.py", line 5, in foobar'
E           and: u'    raise NotImplementedError'
E           and: u''
E           and: u'NotImplementedError'
E           and: u''
E       fnmatch: '*test_doctest_lineno.py:3: UnexpectedException'
E          with: u'/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0/test_doctest_lineno.py:3: UnexpectedException'
E       nomatch: 'Results*:'
E           and: u'\r'
E           and: u'\r \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m \x1b[31m\u2a2f\x1b[0m                                        \x1b[31m100% \x1b[0m\x1b[40m\x1b[31m\u2588\x1b[0m\x1b[40m\x1b[31m\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\x1b[0m'
E           and: u''
E       fnmatch: 'Results*:'
E          with: u'Results (0.01s):'
E       nomatch: '*-*test_doctest_lineno.py*:3*'
E           and: u'\x1b[31m       1 failed\x1b[0m'
E           and: u'         - \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m:2 \x1b[31m[doctest] test_doctest_lineno.foobar\x1b[0m'
E           and: u''
E       remains unmatched: '*-*test_doctest_lineno.py*:3*'

/nix/store/5dzmpz9wapsvbjy4pcslbg32dddgsxxq-python2.7-pytest-3.5.0/lib/python2.7/site-packages/_pytest/pytester.py:1175: Failed
----------------------------- Captured stdout call -----------------------------
Test session starts (platform: linux2, Python 2.7.14, pytest 3.5.0, pytest-sugar 0.9.1)\nrootdir: /tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0, inifile:\nplugins: sugar-0.9.1\n\n\n\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015 [doctest] test_doctest_lineno.foobar \u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\u2015\n002 \n003                 >>> foobar()\nUNEXPECTED EXCEPTION: NotImplementedError()\nTraceback (most recent call last):\n\n  File "/nix/store/l9j2jsc9flrbmpf799nw9wdq59gpwms8-python-2.7.14/lib/python2.7/doctest.py", line 1315, in __run\n    compileflags, 1) in test.globs\n\n  File "<doctest test_doctest_lineno.foobar[0]>", line 1, in <module>\n\n  File "/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0/test_doctest_lineno.py", line 5, in foobar\n    raise NotImplementedError\n\nNotImplementedError\n\n/tmp/nix-build-python2.7-pytest-sugar-0.9.1.drv-0/pytest-of-nixbld1/pytest-0/test_doctest_lineno0/test_doctest_lineno.py:3: UnexpectedException\n\r\n\r \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m \x1b[31m\u2a2f\x1b[0m                                        \x1b[31m100% \x1b[0m\x1b[40m\x1b[31m\u2588\x1b[0m\x1b[40m\x1b[31m\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\x1b[0m\n\nResults (0.01s):\n\x1b[31m       1 failed\x1b[0m\n         - \x1b[36m\x1b[0mtest_doctest_lineno.py\x1b[0m:2 \x1b[31m[doctest] test_doctest_lineno.foobar\x1b[0m
================ 1 failed, 76 passed, 3 skipped in 1.31 seconds ================
builder for ‘/nix/store/xjc1ywcjw2ndy6m2lspynq4hjldwsajy-python2.7-pytest-sugar-0.9.1.drv’ failed with exit code 1
error: build of ‘/nix/store/xjc1ywcjw2ndy6m2lspynq4hjldwsajy-python2.7-pytest-sugar-0.9.1.drv’ failed
```

## Technical details

```
adam: [~] $ nix-shell -p nix-info --run "nix-info -m" -I .nix-channels 
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.15_1, void, noversion`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(adam): `"nixpkgs-18.09pre137305.d91caac6c3e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/adam/channels/nixpkgs`
```

--- a/pkgs/development/python-modules/cheroot/default.nix
+++ b/pkgs/development/python-modules/cheroot/default.nix
@@ -1,6 +1,6 @@
 { stdenv, fetchPypi, buildPythonPackage
 , more-itertools, six
-, coverage, codecov, pytest, pytestcov, pytest-sugar, portend
+, pytest, pytestcov, portend
 , backports_unittest-mock, setuptools_scm }:
 
 buildPythonPackage rec {
@@ -16,7 +16,7 @@ buildPythonPackage rec {
 
   buildInputs = [ setuptools_scm ];
 
-  checkInputs = [ coverage codecov pytest pytestcov pytest-sugar portend backports_unittest-mock  ];
+  checkInputs = [ pytest pytestcov portend backports_unittest-mock ];
 
   checkPhase = ''
     py.test cheroot

--- a/pkgs/development/python-modules/cherrypy/default.nix
+++ b/pkgs/development/python-modules/cherrypy/default.nix
@@ -1,11 +1,10 @@
 { lib, buildPythonPackage, fetchPypi
 , cheroot, portend, routes, six
 , setuptools_scm
-, backports_unittest-mock, codecov, coverage, objgraph, pathpy, pytest, pytest-sugar, pytestcov
+, backports_unittest-mock, objgraph, pathpy, pytest, pytestcov
 }:
 
 buildPythonPackage rec {
-  name = "${pname}-${version}";
   pname = "CherryPy";
   version = "14.0.1";
 
@@ -18,7 +17,7 @@ buildPythonPackage rec {
 
   buildInputs = [ setuptools_scm ];
 
-  checkInputs = [ backports_unittest-mock codecov coverage objgraph pathpy pytest pytest-sugar pytestcov ];
+  checkInputs = [ backports_unittest-mock objgraph pathpy pytest pytestcov ];
 
   checkPhase = ''
     LANG=en_US.UTF-8 pytest

--- a/pkgs/development/python-modules/portend/default.nix
+++ b/pkgs/development/python-modules/portend/default.nix
@@ -1,19 +1,24 @@
 { stdenv, buildPythonPackage, fetchPypi
-, pytest, pytest-sugar, pytest-warnings, setuptools_scm
-, tempora }:
+, pytest, setuptools_scm, tempora }:
 
 buildPythonPackage rec {
-  name = "${pname}-${version}";
   pname = "portend";
   version = "2.2";
-
-  buildInputs = [ pytest pytest-sugar pytest-warnings setuptools_scm ];
-  propagatedBuildInputs = [ tempora ];
 
   src = fetchPypi {
     inherit pname version;
     sha256 = "bc48d3d99e1eaf2e9406c729f8848bfdaf87876cd3560dc3ec6c16714f529586";
   };
+
+  nativeBuildInputs = [ setuptools_scm ];
+
+  propagatedBuildInputs = [ tempora ];
+
+  checkInputs = [ pytest ];
+
+  checkPhase = ''
+    py.test
+  '';
 
   meta = with stdenv.lib; {
     description = "Monitor TCP ports for bound or unbound states";

--- a/pkgs/development/python-modules/pytest-sugar/default.nix
+++ b/pkgs/development/python-modules/pytest-sugar/default.nix
@@ -19,5 +19,9 @@ buildPythonPackage rec {
     description = "A plugin that changes the default look and feel of py.test";
     homepage = https://github.com/Frozenball/pytest-sugar;
     license = licenses.bsd3;
+
+    # incompatible with pytest 3.5
+    # https://github.com/Frozenball/pytest-sugar/issues/134
+    broken = true; # 2018-04-20
   };
 }

